### DISCUSSION TITLE
Gradle (Kotlin) unable to resolve reference nyx

### DISCUSSION THREAD

**Discussion by aharshbarger** [mooltiverse/nyx#96] [General]
I'm running into issues with using the tutorial for a Java project that is using Gradle (Kotlin) DSL.  the issue after adding the plugin to the settings.gradle.kts file and applying, I get  (Unresolved reference: nyx) when trying to use the nyx configuration block.  

Using IntelliJ IDEA 2021.3.1, Gradle Wrapper 7.4, JDK 17 (also tried it with JDK 11)

```
plugins {
    id("com.mooltiverse.oss.nyx") version "1.0.1-fix81.1"
}

nyx {
    // define configurations here
}
```
Any thoughts on what I may be doing wrong?

UPDATE: If I use the `nyx.yml` file instead, it works as expected.   I'll create an issue ticket for this usage scenario.

**Comment by flelli** [comment]
Applying the plugin and configuration in `build.gradle` runs with no issues.

The problem seems to be limited to using it as a settings plugin (in settings.gradle.kts). I'm still investigating this behavior but in the meanwhile using the `build.gradle` might be an option.

**Comment by flelli** [comment]
I managed to have the plugin working with Kotlin both as a _project_ and as a _settings_ plugin.

The _project plugin_ configuration (`build.gradle.kts`) is:

```kotlin
plugins {
  id("com.mooltiverse.oss.nyx") version "1.3.4"
}

nyx {
  dryRun.set(false)
  resume.set(false)
  stateFile.set(".nyx-state.yml")
  verbosity.set("DEBUG")
}
```

While the _settings plugin_ configuration (`settings.gradle.kts`) is:

```kotlin
plugins {
  id("com.mooltiverse.oss.nyx") version "1.3.4"
}

configure<com.mooltiverse.oss.nyx.gradle.NyxExtension> {
  dryRun.set(false)
  resume.set(false)
  stateFile.set(".nyx-state.yml")
  verbosity.set("DEBUG")
}
```

I also added these to the documentation.

Again, I'm no Kotlin expert so other suggestions and contributions are welcome.
